cmake/OpenCVCompilerOptions.cmake

```diff
@@ -196,23 +165,30 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   endif()
 
   if(ENABLE_INSTRUMENTATION)
-    set(OPENCV_EXTRA_CXX_FLAGS "${OPENCV_EXTRA_CXX_FLAGS} --std=c++11")
+    if(NOT " ${CMAKE_CXX_FLAGS} ${OPENCV_EXTRA_CXX_FLAGS}" MATCHES "-std=")
```

Comment:
Maybe we should check `HAVE_CXX11` variable here and produce a warning about incompatible build mode if it is `FALSE`, shouldn't we?